Kickstarter-iOS/Views/Controllers/UpdatePledgeViewController.swift

```diff
@@ -0,0 +1,207 @@
+    .compact()
+
+    arrangedSubviews.forEach { view in
+      self.rootStackView.addArrangedSubview(view)
+    }
+
+    childViewControllers.forEach { viewController in
+      self.addChild(viewController)
+      viewController.didMove(toParent: self)
+    }
+  }
+
+  private func setupConstraints() {
+    NSLayoutConstraint.activate([
+      self.rootStackView.widthAnchor.constraint(equalTo: self.rootScrollView.widthAnchor),
+      self.confirmButton.heightAnchor.constraint(greaterThanOrEqualToConstant: Styles.minTouchSize.height)
+    ])
+  }
+
+  // MARK: - Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self.view
+      |> checkoutBackgroundStyle
+
+    _ = self.rootScrollView
+      |> rootScrollViewStyle
+
+    _ = self.rootStackView
+      |> checkoutRootStackViewStyle
+
+    _ = self.confirmButton
+      |> greenButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.Confirm() }
+
+    _ = self.confirmationLabel
+      |> \.numberOfLines .~ 0
+      |> checkoutBackgroundStyle
+  }
+
+  // MARK: - View model
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.configureWithData
+      .observeForUI()
+      .observeValues { [weak self] data in
+        self?.pledgeAmountViewController.configureWith(value: data)
+        self?.shippingLocationViewController.configureWith(value: data)
+      }
+
+    self.viewModel.outputs.configureSummaryViewControllerWithData
+      .observeForUI()
+      .observeValues { [weak self] project, pledgeTotal in
+        self?.summaryViewController.configureWith(value: (project, pledgeTotal))
+      }
+
+    Keyboard.change
+      .observeForUI()
+      .observeValues { [weak self] change in
+        self?.rootScrollView.handleKeyboardVisibilityDidChange(change)
+      }
+
+    self.shippingLocationViewController.view.rac.hidden
+      = self.viewModel.outputs.shippingLocationViewHidden
+
+    self.confirmationLabel.rac.attributedText = self.viewModel.outputs.confirmationLabelAttributedText
+  }
+
+  // MARK: - Actions
+
+  @objc private func dismissKeyboard() {
+    self.view.endEditing(true)
+  }
+}
+
+// MARK: - PledgeAmountViewControllerDelegate
+
+extension UpdatePledgeViewController: PledgeAmountViewControllerDelegate {
+  func pledgeAmountViewController(
+    _: PledgeAmountViewController,
+    didUpdateAmount amount: Double
+  ) {
+    self.viewModel.inputs.pledgeAmountDidUpdate(to: amount)
+  }
+}
+
+// MARK: - PledgeShippingLocationViewControllerDelegate
+
+extension UpdatePledgeViewController: PledgeShippingLocationViewControllerDelegate {
+  func pledgeShippingLocationViewController(
+    _: PledgeShippingLocationViewController,
+    didSelect shippingRule: ShippingRule
+  ) {
+    self.viewModel.inputs.shippingRuleSelected(shippingRule)
+  }
+}
+
+// MARK: - Styles
+
+private let rootScrollViewStyle: ScrollStyle = { scrollView in
+  scrollView
+    |> UIScrollView.lens.showsVerticalScrollIndicator .~ false
```

Comment:
I was actually going to bring this up ... I think we should be showing scroll indicators or at least showing when the keyboard is on ... because of the low contrast of our background, etc. .. it might be very hard to realize scrolling is an option or where on the page the user is. (does not have to be part of this PR)